tsdb/engine/tsm1/cache.go

```diff
@@ -112,11 +113,15 @@ type Cache struct {
 
 // NewCache returns an instance of a cache which will use a maximum of maxSize bytes of memory.
 // Only used for engine caches, never for snapshots
-func NewCache(maxSize uint64, path string) *Cache {
+func NewCache(maxSize uint64, config tsdb.ShardConfig) *Cache {
```

Comment:
This might be confusing to pass in the shard config to this function since only one of the two attributes included in the shard config are actually used. When using this function, you may wonder "why does it only care about one part of the config?"

I think it might be better to continue passing the path as a string directly here and to modify the call site in `NewEngine` to pass in the relevant string from the shard config. It will also make the diff a lot smaller and won't change the public interface.